server/src/main/java/org/opensearch/search/aggregations/bucket/filterrewrite/FilterRewriteOptimizationContext.java

```diff
@@ -123,12 +136,43 @@ public boolean tryOptimize(final LeafReaderContext leafCtx, final BiConsumer<Lon
         Ranges ranges = getRanges(leafCtx, segmentMatchAll);
         if (ranges == null) return false;
 
-        consumeDebugInfo(aggregatorBridge.tryOptimize(values, incrementDocCount, ranges));
+        Supplier<DocIdSetBuilder> disBuilderSupplier = null;
+        if (subAggLength != 0) {
+            disBuilderSupplier = () -> {
+                try {
+                    return new DocIdSetBuilder(leafCtx.reader().maxDoc(), values, aggregatorBridge.fieldType.name());
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
```

Comment:
Why are we catching a checked exception and rethrowing an unchecked exception here when the method is documented to throw IOException?

Reply:
The idea is to abort the optimization if this IOException happens somehow.
This IOException cannot bubble up because it's used in a interface method (grow() of PointValues) that doesn't throw IOException. So we catch and throw a different one.